src/frame/decompress.rs

```diff
@@ -399,3 +398,31 @@ impl<R: fmt::Debug + io::Read> fmt::Debug for FrameDecoder<R> {
             .finish()
     }
 }
+
+/// Similar to set_len but panics if the vec doesn't have enough capacity.
```

Comment:
this method doesn't panic

Reply:
The unsafe version does :thinking:  Should l replace the debug_assert with a plain assert?

Reply:
I would just adjust the comments